features/cryptocell/FEATURE_CRYPTOCELL310/ccm_alt.c

```diff
@@ -84,8 +84,11 @@ int mbedtls_ccm_encrypt_and_tag( mbedtls_ccm_context *ctx, size_t length,
     if( iv_len < 7 || iv_len > 13 )
         return ( MBEDTLS_ERR_CCM_BAD_INPUT );
 
+    if( add_len > 0xFF00 )
```

Comment:
There is something off: commit message says `tag_len` and the `add_len` is being checked here. What is the intention with this check?

Reply:
It's a typo in the commit message. I'll amend the message

Reply:
Before you do that, could you please explain why this check is necessary?

Reply:
It was raised by the [test case](https://github.com/ARMmbed/mbedtls/blob/development/tests/suites/test_suite_ccm.data#L38)
And check taken from the [Mbed TLS code](https://github.com/ARMmbed/mbedtls/blob/development/library/ccm.c#L165)

Reply:
This is an Mbed TLS specific limitation, we shouldn't constrain hardware accelerators that might have full support. We should run this test only for the Mbed TLS implementation:
https://github.com/ARMmbed/mbedtls/issues/1996

Reply:
Thanks, I'll remove this commit from the PR